src/frontend/src/components/forms/fields/RelatedModelField.tsx

```diff
@@ -0,0 +1,203 @@
+import { t } from '@lingui/macro';
+import { Input } from '@mantine/core';
+import { UseFormReturnType } from '@mantine/form';
+import { useDebouncedValue } from '@mantine/hooks';
+import { useQuery } from '@tanstack/react-query';
+import { ReactNode, useEffect, useMemo, useState } from 'react';
+import Select from 'react-select';
+
+import { api } from '../../../App';
+import { RenderInstance } from '../../render/Instance';
+import { ApiFormProps } from '../ApiForm';
+import { ApiFormFieldSet, ApiFormFieldType } from './ApiFormField';
+import { constructField } from './ApiFormField';
+
+/**
+ * Render a 'select' field for searching the database against a particular model type
+ */
+export function RelatedModelField({
+  error,
+  formProps,
+  form,
+  fieldName,
+  field,
+  definitions,
+  limit = 10
+}: {
+  error: ReactNode;
+  formProps: ApiFormProps;
+  form: UseFormReturnType<Record<string, unknown>>;
+  field: ApiFormFieldType;
+  fieldName: string;
+  definitions: ApiFormFieldSet;
+  limit?: number;
+}) {
+  // Extract field definition from provided data
+  // Where user has provided specific data, override the API definition
+  const definition: ApiFormFieldType = useMemo(
+    () =>
+      constructField({
+        form: form,
+        field: field,
+        fieldName: fieldName,
+        definitions: definitions
+      }),
+    [form.values, field, definitions]
+  );
+
+  // Keep track of the primary key value for this field
+  const [pk, setPk] = useState<number | null>(null);
+
+  // If an initial value is provided, load from the API
+  useEffect(() => {
+    // If a value is provided, load the related object
+    if (form.values) {
+      let formPk = form.values[fieldName];
+
+      if (formPk && formPk != pk) {
+        let url = (definition.api_url || '') + formPk + '/';
+
+        // TODO: Fix this!!
+        if (url.startsWith('/api')) {
+          url = url.substring(4);
+        }
+
+        api.get(url).then((response) => {
+          let data = response.data;
+
+          if (data && data.pk) {
+            let value = {
+              value: data.pk,
+              data: data
+            };
+
+            setData([value]);
+            setPk(data.pk);
+          }
+        });
+      }
+    }
+  }, [form]);
+
+  const [offset, setOffset] = useState<number>(0);
+
+  const [data, setData] = useState<any[]>([]);
+
+  // Search input query
+  const [value, setValue] = useState<string>('');
+  const [searchText, cancelSearchText] = useDebouncedValue(value, 250);
+
+  const queryController = new AbortController();
```

Comment:
You know, that this will be a different class instance on each render? You would need to use a useRef

Reply:
Good catch, I've added a hook around that :)

Reply:
Actually, if you need this more often, you could think about making your own hook around this like a useAboardControler so you don't have to duplicate code.